sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/With.scala

```diff
@@ -92,6 +95,21 @@ object With {
     val commonExprRefs = commonExprDefs.map(new CommonExpressionRef(_))
     With(replaced(commonExprRefs), commonExprDefs)
   }
+
+  private[sql] def childContainsUnsupportedAggExpr(withExpr: With): Boolean = {
+    lazy val commonExprIds = withExpr.defs.map(_.id).toSet
+    withExpr.child.exists {
+      case agg: AggregateExpression =>
```

Comment:
nit: ideally we can stop the traversal after hitting `AggregateExpression` because there can't be  nested `AggregateExpression`. This requires to write a manual recursion instead of using `.exist`.

Reply:
i agree, or we could add some pruning-based tree traversal functions like `existsWithPruning` (similar to `transformWithPruning`)